src/components/call/CallGeneralInfo.tsx

```diff
@@ -53,6 +74,62 @@ const CallGeneralInfo: React.FC<{
     }
   }, [startCall, endCall, formik]);
 
+  function validateRefnumber(input: string) {
+    let error;
+    const regExp = /((([a-z|\d]+)){)({digits:[1-9]+})/;
+    const prefix = /((([a-z|\d]+)){)/;
+    const suffix = /({digits:[1-9]+})/;
+    if (input) {
+      if (input.match(regExp)) {
+      } else if (!input.match(prefix)) {
+        error = 'Invalid prefix.';
+      } else if (!input.match(suffix)) {
+        error = 'Invalid parameter.';
+      }
+    }
+
+    return error;
+  }
```

Comment:
Thanks for clarifying about the error messages 🙂.
Perhaps we could simplify by removing the `regExp` bit? (it seems the sub-regexs do all the necessary checks).
I've also fleshed out the variable names for a bit more clarity (but I'll leave the final decision with you):
```suggestion
  function validateRefnumber(input: string) {
    let errorMessage;
    const prefixRegex = /((([a-z|\d]+)){)/;
    const parameterRegex = /({digits:[1-9]+})/;
    if (input) {
      if (!input.match(prefixRegex)) {
        errorMessage = 'Invalid prefix.';
      } else if (!input.match(parameterRegex)) {
        errorMessage = 'Invalid parameter.';
      }
    }
    return errorMessage;
  }
```
**Again, please check it works before committing** 😁

Reply:
I think a main regex might still be necessary @JEThompson97, or the user could theoretically have some whitespace between the prefix and parameter and the check would still pass. I like the naming though, that seems clearer 😊

Some possible improvements to the regex part: 

- Might be a good idea to include the `^` and `$` characters for the start and end of string to the main regex. This will avoid the user adding whitespace before and after an otherwise valid format.
- It looks like in the full regular expression you are defining groups in there (the brackets), so I think you could get rid of the prefix and suffix regexes and instead check the indices of the match (https://javascript.info/regexp-groups#parentheses-contents-in-the-match) to check the prefix and parameter groups (see example below). This will be more efficient because you're only executing one match. There is a nicer way to refer to the groups by given names instead of indices, but I know we struggled to get this working when we had a look @RasmiaKulan. 

- Instead of doing nothing when the input matches the regex, it would be better practice (i.e. more explicit and easier to read) if you invert the condition and then do the additional checks 
  ```javascript
  const match = input.match(regExp);
  if (!match[0]) {
    if (!match[1]) {
      error = 'Invalid prefix';
    } else if (!match[2]) {
      error = 'Invalid parameter';
    } else {
      error = 'Invalid format';
  }
  ```

- Have an else condition that is just 'Invalid format' (as above), for when the input doesn't match the regex, but does match the prefix and parameter. I think this input be an example of this: `192      {digits:4}` (whitespace between the capture groups)
- Capitalise the N so that it's `validateRefNumber`